debian/rules

```diff
@@ -93,7 +93,7 @@ DEPENDS += pam-challenge-response, \
 # Platform-specific dependencies
 DEPENDS.aws = nvme-cli,
 DEPENDS.azure = walinuxagent,
-DEPENDS.esx = open-vm-tools,
+DEPENDS.esx = open-vm-tools, ca-certificates-java,
```

Comment:
is this package specific to ESX? IMO, we should try to minimize the differences between the various cloud platforms, so I'd like to install this package in all cases, rather than specific to ESX, if possible.

Reply:
This package is only consumed by a variant that is only used by ESX. As discussed in [this Slack thread in #dcol-dev](https://delphix.slack.com/archives/CMM9L7SG0/p1627667213018900), this package is only consumed by the `internal-dcenter` variant, whose only consumer is ESX. To keep package installations as minimal as possible I decided to only install it on ESX. In addition, as part of [TOOL-11962](https://jira.delphix.com/browse/TOOL-11962) I plan to deprecate the ability to build the `internal-dcenter` variant for any platform aside from ESX. Doing so will decrease the number of appliance-build-stage1 jobs from 42 (6 variants * 7 platforms) to 36 (5 variants * 7 platforms).

Reply:
> To keep package installations as minimal as possible I decided to only install it on ESX.

Please correct me if I'm mistaken, but I interpret this to mean that we could install this package on all platforms without adverse effects. Thus, my preference would be to do that. IMO, I value consistency of the product across the cloud platforms, even when that isn't strictly necessary, than intentional inconsistency to avoid installing packages on platforms that may not need the package.

With that said, I'm not sure what our "official" stance is w.r.t. cloud platform consistency, so others may disagree.

> In addition, as part of TOOL-11962 I plan to deprecate the ability to build the internal-dcenter variant for any platform aside from ESX.

Sure, that's fine, but tangential to this specific issue/PR. For example, this new package will not be installed on non-dcenter variants, so I just want to make sure this change is "correct" for the non-dcenter cases as well.

Reply:
I'd tend to agree with Prakash on that. Unless `ca-certificates-java` causes some unwanted side-effects on other platforms or takes an excessive amount of space I'd say it's preferable to install it on all platforms. Moreover, given that most of our automated testing is run on AWS, if that package does cause some side effects we'd be more likely to catch it this way.

Reply:
Thank you both for your feedback. I'll install the package on all platforms and re-build with all testing.